storage/bolt/share.go

```diff
@@ -75,3 +75,11 @@ func (s shareBackend) Delete(hash string) error {
 	}
 	return err
 }
+
+func (s shareBackend) DeleteWithPathPrefix(pathPrefix string) {
+	var links []share.Link
+	s.db.Prefix("Path", pathPrefix, &links)
+	for _, link := range links {
+		s.db.DeleteStruct(&share.Link{Hash: link.Hash})
+	}
+}
```

Comment:
We should check for errors. Maybe something like this:

```suggestion
func (s shareBackend) DeleteWithPathPrefix(pathPrefix string) error {
	var links []share.Link
	if err := s.db.Prefix("Path", pathPrefix, &links); err != nil {
		return err
	}
	var err error
	for _, link := range links {
		err = errors.Join(err, s.db.DeleteStruct(&share.Link{Hash: link.Hash}))
	}
	return err
}
```

And also propagate the error back to the delete handler. There, I wouldn't return a 500, but instead just log it as a warning.

Reply:
Right. Joining errors is something I considered but didn't know how to handle them, and decided to not do anything initially since it failing is not fatal in regards to the file deletion handler. Sorry for this back and forth though... Rebased and updated!